test/library/postman-api-key.test.js

```diff
@@ -0,0 +1,207 @@
+                .that.has.property('X-Api-Key', '12345678');
+
+            expect(summary).to.be.an('object')
+                .that.has.property('collection').to.be.an('object')
+                .and.include({ id: 'C1', name: 'Collection' });
+
+            expect(summary.run.failures).to.be.empty;
+            expect(summary.run.executions, 'should have 1 execution').to.have.lengthOf(1);
+
+            done();
+        });
+    });
+
+    it('should fetch environment via UID', function (done) {
+        newman.run({
+            collection: 'test/fixtures/run/single-get-request.json',
+            environment: '1234-931c1484-fd1e-4ceb-81d0-2aa102ca8b5f',
+            postmanApiKey: '12345678'
+        }, function (err, summary) {
+            expect(err).to.be.null;
+            sinon.assert.calledOnce(request.get);
+
+            let requestArg = request.get.firstCall.args[0];
+
+            expect(requestArg).to.be.an('object').with.keys(['url', 'json', 'headers']);
+
+            expect(requestArg.url)
+                .to.equal('https://api.getpostman.com/environments/1234-931c1484-fd1e-4ceb-81d0-2aa102ca8b5f');
+
+            expect(requestArg.headers).to.be.an('object')
+                .that.has.property('X-Api-Key', '12345678');
+
+            expect(summary).to.be.an('object')
+                .that.has.property('environment').to.be.an('object')
+                .and.include({ id: 'E1', name: 'Environment' });
+
+            expect(summary.run.failures).to.be.empty;
+            expect(summary.run.executions, 'should have 1 execution').to.have.lengthOf(1);
+
+            done();
+        });
+    });
+
+    it('should fetch all resources via UID', function (done) {
+        newman.run({
+            collection: '1234-588025f9-2497-46f7-b849-47f58b865807',
+            environment: '1234-931c1484-fd1e-4ceb-81d0-2aa102ca8b5f',
+            postmanApiKey: '12345678'
+        }, function (err, summary) {
+            expect(err).to.be.null;
+            sinon.assert.calledTwice(request.get);
+
+            expect(summary).to.be.an('object').with.keys(['collection', 'environment', 'globals', 'run']);
+
+            expect(summary.collection).to.include({ id: 'C1', name: 'Collection' });
+            expect(summary.environment).to.include({ id: 'E1', name: 'Environment' });
+
+            expect(summary.run.failures).to.be.empty;
+            expect(summary.run.executions, 'should have 1 execution').to.have.lengthOf(1);
+
+            done();
+        });
+    });
+
+    it('should throw error without postmanApiKey', function (done) {
+        newman.run({
+            collection: '1234-588025f9-2497-46f7-b849-47f58b865807'
+        }, function (err) {
+            expect(err).to.be.ok.that.match(/no such file or directory/);
+            sinon.assert.notCalled(request.get);
+
+            done();
+        });
+    });
+
+    it('should not pass API Key header with URL ', function (done) {
```

Comment:
Test title is incomplete. Also, add a test (if not already present) for cases where the API Key should be sent with regular Postman API URls.